src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/FrameHandling/FrameIterator.cs

```diff
@@ -0,0 +1,163 @@
+                return;
+
+            case FrameType.FuncEvalFrame:
+                Data.FuncEvalFrame funcEvalFrame = target.ProcessedData.GetOrAdd<Data.FuncEvalFrame>(CurrentFrame.Address);
+                GetFrameHandler(context).HandleFuncEvalFrame(funcEvalFrame);
+                return;
+
+            // ResumableFrame type frames
+            case FrameType.ResumableFrame:
+            case FrameType.RedirectedThreadFrame:
+                Data.ResumableFrame resumableFrame = target.ProcessedData.GetOrAdd<Data.ResumableFrame>(CurrentFrame.Address);
+                GetFrameHandler(context).HandleResumableFrame(resumableFrame);
+                return;
+
+            case FrameType.FaultingExceptionFrame:
+                Data.FaultingExceptionFrame faultingExceptionFrame = target.ProcessedData.GetOrAdd<Data.FaultingExceptionFrame>(CurrentFrame.Address);
+                GetFrameHandler(context).HandleFaultingExceptionFrame(faultingExceptionFrame);
+                return;
+
+            case FrameType.HijackFrame:
+                Data.HijackFrame hijackFrame = target.ProcessedData.GetOrAdd<Data.HijackFrame>(CurrentFrame.Address);
+                GetFrameHandler(context).HandleHijackFrame(hijackFrame);
+                return;
+            default:
+                // Unknown Frame type. This could either be a Frame that we don't know how to handle,
+                // or a Frame that does not update the context.
+                return;
+        }
+    }
+
+    public bool IsInlineCallFrameWithActiveCall()
+    {
+        if (GetFrameType(CurrentFrame) != FrameType.InlinedCallFrame)
+        {
+            return false;
+        }
+        Data.InlinedCallFrame inlinedCallFrame = target.ProcessedData.GetOrAdd<Data.InlinedCallFrame>(currentFramePointer);
+        return inlinedCallFrame.CallerReturnAddress != 0;
+    }
+
+    private FrameType GetFrameType(Data.Frame frame)
+    {
+        foreach (FrameType frameType in Enum.GetValues<FrameType>())
+        {
+            TargetPointer typeVptr;
+            try
+            {
+                // not all Frames are in all builds, so we need to catch the exception
+                typeVptr = target.ReadGlobalPointer(frameType.ToString() + "Identifier");
+                if (frame.Identifier == typeVptr)
+                {
+                    return frameType;
+                }
+            }
+            catch (InvalidOperationException)
+            {
+            }
+        }
+
+        return FrameType.Unknown;
+    }
+
+    private IPlatformFrameHandler GetFrameHandler(IPlatformAgnosticContext context)
+    {
+        return context switch
+        {
+            ContextHolder<AMD64Context> contextHolder => new AMD64FrameHandler(target, contextHolder),
+            ContextHolder<ARM64Context> contextHolder => new ARM64FrameHandler(target, contextHolder),
+            _ => throw new InvalidOperationException("Unsupported context type"),
```

Comment:
Might be nice to say which type is unsupported, it will make failures in production more actionable. Not strictly necessary though.